File: src/main/java-graphql/io/github/graphql/model/RegistryPackageFileEdgeResponseProjection.java
package io.github.graphql.model;

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseField;
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseProjection;

/**
 * Response projection for RegistryPackageFileEdge
 */
@javax.annotation.Generated(
    value = "com.kobylynskyi.graphql.codegen.GraphQLCodegen",
    date = "2021-03-06T19:47:47+0800"
)
public class RegistryPackageFileEdgeResponseProjection extends GraphQLResponseProjection {

    public RegistryPackageFileEdgeResponseProjection() {
    }

    @Override
    public RegistryPackageFileEdgeResponseProjection all$() {
        return all$(3);
    }

    @Override
    public RegistryPackageFileEdgeResponseProjection all$(int maxDepth) {
        this.cursor();
        if (projectionDepthOnFields.getOrDefault("RegistryPackageFileEdgeResponseProjection.RegistryPackageFileResponseProjection.node", 0) <= maxDepth) {
            projectionDepthOnFields.put("RegistryPackageFileEdgeResponseProjection.RegistryPackageFileResponseProjection.node", projectionDepthOnFields.getOrDefault("RegistryPackageFileEdgeResponseProjection.RegistryPackageFileResponseProjection.node", 0) + 1);
            this.node(new RegistryPackageFileResponseProjection().all$(maxDepth - projectionDepthOnFields.getOrDefault("RegistryPackageFileEdgeResponseProjection.RegistryPackageFileResponseProjection.node", 0)));
        }
        this.typename();
        return this;
    }

    public RegistryPackageFileEdgeResponseProjection cursor() {
        return cursor(null);
    }

    public RegistryPackageFileEdgeResponseProjection cursor(String alias) {
        fields.add(new GraphQLResponseField("cursor").alias(alias));
        return this;
    }

    public RegistryPackageFileEdgeResponseProjection node(RegistryPackageFileResponseProjection subProjection) {
        return node(null, subProjection);
    }

    public RegistryPackageFileEdgeResponseProjection node(String alias, RegistryPackageFileResponseProjection subProjection) {
        fields.add(new GraphQLResponseField("node").alias(alias).projection(subProjection));
        return this;
    }

    public RegistryPackageFileEdgeResponseProjection typename() {
        return typename(null);
    }

    public RegistryPackageFileEdgeResponseProjection typename(String alias) {
        fields.add(new GraphQLResponseField("__typename").alias(alias));
        return this;
    }


}

File: src/main/kotlin-graphql/io/github/graphql/kotlin/model/RegistryPackageFileEdgeResponseProjection.kt
package io.github.graphql.kotlin.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseField
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseProjection
import java.util.Objects

/**
 * Response projection for RegistryPackageFileEdge
 */
@javax.annotation.Generated(
    value = ["com.kobylynskyi.graphql.codegen.GraphQLCodegen"],
    date = "2021-03-07T19:48:13+0800"
)
open class RegistryPackageFileEdgeResponseProjection : GraphQLResponseProjection() {

    override fun `all$`(): RegistryPackageFileEdgeResponseProjection = `all$`(3)

    override fun `all$`(maxDepth: Int): RegistryPackageFileEdgeResponseProjection {
        this.cursor()
        if (projectionDepthOnFields.getOrDefault("RegistryPackageFileEdgeResponseProjection.RegistryPackageFileResponseProjection.node", 0) <= maxDepth) {
            projectionDepthOnFields["RegistryPackageFileEdgeResponseProjection.RegistryPackageFileResponseProjection.node"] = projectionDepthOnFields.getOrDefault("RegistryPackageFileEdgeResponseProjection.RegistryPackageFileResponseProjection.node", 0) + 1
            this.node(RegistryPackageFileResponseProjection().`all$`(maxDepth - projectionDepthOnFields.getOrDefault("RegistryPackageFileEdgeResponseProjection.RegistryPackageFileResponseProjection.node", 0)))
        }
        this.typename()
        return this
    }

    fun cursor(): RegistryPackageFileEdgeResponseProjection = cursor(null)

    fun cursor(alias: String?): RegistryPackageFileEdgeResponseProjection {
        fields.add(GraphQLResponseField("cursor").alias(alias))
        return this
    }

    fun node(subProjection: RegistryPackageFileResponseProjection): RegistryPackageFileEdgeResponseProjection = node(null, subProjection)

    fun node(alias: String?, subProjection: RegistryPackageFileResponseProjection): RegistryPackageFileEdgeResponseProjection {
        fields.add(GraphQLResponseField("node").alias(alias).projection(subProjection))
        return this
    }

    fun typename(): RegistryPackageFileEdgeResponseProjection = typename(null)

    fun typename(alias: String?): RegistryPackageFileEdgeResponseProjection {
        fields.add(GraphQLResponseField("__typename").alias(alias))
        return this
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || javaClass != other.javaClass) {
            return false
        }
        val that = other as RegistryPackageFileEdgeResponseProjection
        return Objects.equals(fields, that.fields)
    }

    override fun hashCode(): Int = Objects.hash(fields)

}
